### PR TITLE
Add date picker to SightingEditScreen

### PR DESCRIPTION
**Pull request by Jonri2**
The Date Picker works differently for iOS and Android so I did my best to test on both, but there may be some platform-specific issues.

### CODE REVIEW

File: backend/index.js [hunk 1]
```diff
@@ -21,7 +21,6 @@ app.get('/', (request, response) => {
 app.all('*', function (req, res, next) {
   let token = getBearerToken(req)
   if (token == SECRET) {
-    next() // Go onto the queries.
```
Comment (Jonri2) [inline backend/index.js:24]: This fixes an issue where the first request always fails. I added it to this pull request just to make it simpler to review.

File: backend/index.js [hunk 2]
```diff
@@ -41,6 +40,7 @@ app.all('*', function (req, res, next) {
     app.get('/photo/turtle/:turtleId', db.getPhotoByTurtleId)
     app.get('/photo/sighting/:sightingId', db.getPhotoBySightingId)
     app.get('/email/:address', db.sendEmail)
+    next() // Go onto the queries.
```
Comment (Jonri2) [inline backend/index.js:43]: This fixes an issue where the first request always fails. I added it to this pull request just to make it simpler to review.

Comment (charkour) [reply backend/index.js:43]: Thanks! That bug is so annoying. Why do you think the first one always fails?

Comment (Jonri2) [reply backend/index.js:43]: I think that when the first request came, next() would be called before the routes were defined, so it wouldn't be routed but the routes would still get defined after a response was sent. Then, on the next call, the routes would have already been defined from the previous call so there wouldn't be an issue.

Comment (charkour) [reply backend/index.js:43]: Ah makes sense. Thanks!

**Pull request merged**